onnxruntime/contrib_ops/cpu/transformers/beam_search_impl_base.h

```diff
@@ -255,36 +251,69 @@ template <typename T>
 Status BeamSearchBase<T>::GenerateNextToken(
     const OrtValue& logits,
     gsl::span<int32_t>& beam_next_tokens,
-    gsl::span<int32_t>& beam_indices,
     BeamSearchState<T>& beam_state,
     BeamSearchCpuState& cpu_state,
     int counter) {
   // Process logits to get next token scores
   ORT_RETURN_IF_ERROR(ProcessLogits(logits, beam_state, cpu_state, temp_space_allocator_, counter));
 
-  gsl::span<float>& beam_scores = beam_scorer_->GetNextScores();
-  // It is optional to clone beam_scores. Change it to use same buffer also works for CPU:
-  //    beam_state.beam_scores = beam_scores
-  // Here we make a copy to reduce the coupling with little cost (the buffer size is small).
-  ORT_RETURN_IF_ERROR(device_copy_func_(beam_state.beam_scores,
-                                        beam_scores,
-                                        ort_stream_,
-                                        DeviceCopyDirection::hostToDevice));
+  if (this->IsCuda()) {
+    auto beam_scores = beam_scorer_->GetNextScores();
+    // It is optional to clone beam_scores. Change it to use same buffer also works for CPU:
```

Comment:
we can update comments here like remove "for CPU"

Reply:
Or maybe we can replace deviceToDevice copy by `beam_state.beam_scores = beam_scores` here.